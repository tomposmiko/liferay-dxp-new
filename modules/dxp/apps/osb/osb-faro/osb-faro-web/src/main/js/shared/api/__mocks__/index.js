import * as accounts from './accounts';
import * as activities from './activities';
import * as apiTokens from './api-tokens';
import * as assets from './assets';
import * as blockedKeywords from './blocked-keywords';
import * as channels from './channels';
import * as contactsCards from './contacts-cards';
import * as dataSource from './data-source';
import * as definitions from './definitions';
import * as distributions from './distributions';
import * as fieldMappings from './field-mappings';
import * as individuals from './individuals';
import * as individualSegment from './individual-segment';
import * as interests from './interests';
import * as issue from './issue';
import * as notifications from './notifications';
import * as pagesVisited from './pages-visited';
import * as preferences from './preferences';
import * as projects from './projects';
import * as user from './user';

export {
	accounts,
	activities,
	apiTokens,
	assets,
	blockedKeywords,
	channels,
	contactsCards,
	dataSource,
	definitions,
	distributions,
	fieldMappings,
	individualSegment,
	individuals,
	interests,
	issue,
	notifications,
	pagesVisited,
	preferences,
	projects,
	user
};
