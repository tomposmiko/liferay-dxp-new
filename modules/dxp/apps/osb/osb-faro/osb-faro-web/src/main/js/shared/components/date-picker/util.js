import moment from 'moment';
import {conformsTo, isNil} from 'lodash';

/**
 * Represents a time between two dates.
 * @typedef Range
 * @property {Moment} start - The start of the range.
 * @property {Moment} end - The end of the range.
 */

export function isAboveMaxRange(date, maxRange) {
	return (
		isRange(date) &&
		date.start &&
		date.end &&
		maxRange &&
		moment
			.duration({
				from: date.start,
				to: date.end
			})
			.asDays() > maxRange
	);
}

/**
 * A validation function that determines if a value is a valid
 * moment instance, allowing for null.
 * @param {Object} value - The value to test.
 * @returns {boolean}
 */
function isMomentOrNil(value) {
	return isNil(value) || moment.isMoment(value);
}

/**
 * A validation function that determines if the value is a valid
 * moment or range.
 * @param {Object} value - The value to test.
 * @returns {boolean}
 */
export function isDateOrRange(value) {
	return isNil(value) || moment.isMoment(value) || isRange(value);
}

/**
 * Determines if a date is in a {@link Range}.
 * @param {Range|Moment} dateOrRange - The range to check in.
 * @param {Moment} date - The date to check for.
 * @returns {boolean}
 */
export function isInRange(dateOrRange, date) {
	if (moment.isMoment(dateOrRange)) {
		return false;
	}

	const {end, start} = dateOrRange;

	return start && date.isAfter(start) && end && date.isBefore(end);
}

/**
 * A validation function that determines if a value is a valid {@link Range}.
 * @param {Object} value - The value to test.
 * @returns {boolean}
 */
export function isRange(value) {
	return conformsTo(value, {
		end: isMomentOrNil,
		start: isMomentOrNil
	});
}

/**
 * Adds a date to an existing {@link Range}.
 *
 * This function will try to be smart and first update the
 * start or end if either are null. If both exist, then it will return
 * a new range with the start set to the new date, and the end reset
 * to null.
 * @param {Range} range - The range to update.
 * @param {Moment} date - The date to add to the range.
 * @returns {Range}
 */
export function updateRange(range, date) {
	if (!range.start) {
		return {
			...range,
			start: date
		};
	}

	if (range.start && !range.end) {
		if (date.isSameOrAfter(range.start)) {
			return {
				...range,
				end: date
			};
		} else {
			return {
				end: range.start,
				start: date
			};
		}
	}

	if (date.isSame(range.start) && date.isSame(range.end)) {
		return range;
	}

	return {
		end: null,
		start: date
	};
}
