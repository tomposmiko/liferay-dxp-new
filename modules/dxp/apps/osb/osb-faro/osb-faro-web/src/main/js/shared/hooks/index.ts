import {useEffect, useState} from 'react';

export {default as useDeepEqualEffect} from './useDeepEqualEffect';
export {default as useQueryPagination} from './useQueryPagination';
export {default as useQueryParams} from './useQueryParams';
export {default as useQueryRangeSelectors} from './useQueryRangeSelectors';
export {default as useRequest} from './useRequest';
export {default as useStatefulPagination} from './useStatefulPagination';

export const useDebounce = (value: any, delay: number) => {
	const [debouncedValue, setDebouncedValue] = useState(value);

	useEffect(
		() => {
			const handler = setTimeout(() => {
				setDebouncedValue(value);
			}, delay);

			return () => {
				clearTimeout(handler);
			};
		},
		// This is required when the `object` has lost the
		// reference plus the values are the same, `useEffect`
		// uses `Object.is` or equivalent under the covers.
		// For some reason the reference is being lost.
		typeof value === 'object' && value !== null
			? [...Object.keys(value), ...Object.values(value)]
			: [value]
	);

	return debouncedValue;
};

export const useFakeLoading = (data: any, time: number = 125) => {
	const [loading, setLoading] = useState(true);

	useEffect(() => {
		setLoading(true);

		setTimeout(() => setLoading(false), time);
	}, [data]);

	return loading;
};
