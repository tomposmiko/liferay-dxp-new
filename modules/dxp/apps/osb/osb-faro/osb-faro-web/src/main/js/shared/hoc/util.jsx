import ErrorDisplay from 'shared/components/ErrorDisplay';
import ErrorPage from 'shared/pages/ErrorPage';
import LoadingPage from 'shared/pages/Loading';
import NoResultsDisplay from 'shared/components/NoResultsDisplay';
import React from 'react';
import Spinner from 'shared/components/Spinner';
import {compose} from 'redux';
import {get, omit} from 'lodash';
import {sub} from 'shared/util/lang';
import {toRoute} from 'shared/util/router';

/**
 * HOC for ErrorDisplay.
 * @param {Object} - Options object to pass as props to ErrorDisplay
 * @returns {Function} Returns the ErrorDisplay or Wrapped Component.
 */
export const withError =
	(options = {}) =>
	Component =>
	({error, errorProps = {}, pageDisplay = true, refetch, ...otherProps}) => {
		const otherOptions = omit(options, 'page');

		if (error) {
			return get(options, 'page', pageDisplay) ? (
				<ErrorPage {...errorProps} {...otherOptions} />
			) : (
				<ErrorDisplay
					onReload={refetch}
					spacer
					{...errorProps}
					{...otherOptions}
				/>
			);
		}

		return <Component refetch={refetch} {...otherProps} />;
	};

/**
 * HOC for NoResultsDisplay.
 * @param {Object} - Options object to pass as props to NoResultsDisplay.
 * @returns {Function} Returns the NoResultsDisplay or WrappedComponent.
 */
export const withEmpty =
	(options = {}) =>
	Component =>
	({
		data,
		error,
		loading,
		noResultsRenderer,
		noResultsRendererProps,
		total,
		...props
	}) => {
		if (((data && data.total === 0) || total === 0) && !loading && !error) {
			if (noResultsRenderer) {
				const NoResults = noResultsRenderer;

				return <NoResults />;
			}

			return (
				<NoResultsDisplay {...options} {...noResultsRendererProps} />
			);
		}

		return (
			<Component
				data={data}
				error={error}
				loading={loading}
				total={total}
				{...props}
			/>
		);
	};

/**
 * HOC for Loading display.
 * @param {Object} - Options object to pass as props to Loading component.
 * @returns {Function} Returns the Loading or WrappedComponent.
 */
export const withLoading =
	(options = {}) =>
	Component =>
	({
		alignCenter = false,
		className,
		data,
		fadeIn = true,
		inline = false,
		loading,
		pageDisplay = true,
		...otherProps
	}) => {
		if (loading) {
			return get(options, 'page', pageDisplay) ? (
				<LoadingPage
					className={className}
					fadeIn={get(options, 'fadeIn', fadeIn)}
					key='LOADING'
				/>
			) : (
				<Spinner
					alignCenter={get(options, 'alignCenter', alignCenter)}
					className={className}
					inline={get(options, 'inline', inline)}
					key='SPINNER'
					spacer={!get(options, 'inline', inline)}
				/>
			);
		}

		return <Component className={className} data={data} {...otherProps} />;
	};

export const withNull =
	(key, errorProps = {}) =>
	Component =>
	props => {
		const {entityType = Liferay.Language.get('page'), linkRoute} =
			errorProps;

		if (key && !props[key]) {
			return (
				<ErrorPage
					{...props}
					href={toRoute(linkRoute, props.router.params)}
					linkLabel={sub(Liferay.Language.get('go-to-x'), [
						entityType
					])}
					message={sub(
						Liferay.Language.get(
							'the-x-you-are-looking-for-does-not-exist'
						),
						[entityType.toLowerCase()]
					)}
					subtitle={sub(Liferay.Language.get('x-not-found'), [
						entityType
					])}
					title={Liferay.Language.get('404')}
				/>
			);
		}

		return <Component {...props} />;
	};

/**
 * HOC for displaying results.
 */
export const SafeResults = compose(
	withLoading(),
	withError()
)(({children, data}) => children(data));

export const WrapSafeResults = compose(
	withLoading(),
	withError()
)(({children}) => children);
