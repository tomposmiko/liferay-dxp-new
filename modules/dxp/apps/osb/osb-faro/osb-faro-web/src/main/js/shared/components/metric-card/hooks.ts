import gql from 'graphql-tag';
import {fetchPolicyDefinition} from 'shared/util/graphql';
import {Filters, getFilters, RawFilters} from 'shared/util/filter';
import {getSafeRangeSelectors} from 'shared/util/util';
import {Interval, RangeSelectors, SafeRangeSelectors} from 'shared/types';
import {useParams} from 'react-router-dom';
import {useQuery} from '@apollo/react-hooks';

interface ICommonVariables extends SafeRangeSelectors, Filters {
	interval: Interval;
}

export const useAssetVariables = (commonVariables: ICommonVariables) => {
	const {assetId, channelId, title, touchpoint} = useParams();

	return {
		assetId: decodeURIComponent(assetId),
		channelId,
		title: decodeURIComponent(title),
		touchpoint:
			touchpoint !== 'Any' ? decodeURIComponent(touchpoint) : null,
		...commonVariables
	};
};

type TMetricQuery = {
	filters: RawFilters;
	interval: Interval;
	Query: typeof gql;
	rangeSelectors: RangeSelectors;
	variables: (commonVariables: ICommonVariables) => any;
};

export const useMetricQuery = ({
	Query,
	filters,
	interval,
	rangeSelectors,
	variables
}: TMetricQuery) => {
	const {data, error, loading} = useQuery(Query, {
		fetchPolicy: fetchPolicyDefinition(rangeSelectors),
		variables: variables({
			interval,
			...getFilters(filters),
			...getSafeRangeSelectors(rangeSelectors)
		})
	});

	return {data, error, loading};
};
