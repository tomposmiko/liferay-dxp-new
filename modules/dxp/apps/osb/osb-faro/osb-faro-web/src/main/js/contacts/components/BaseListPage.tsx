import * as breadcrumbs from 'shared/util/breadcrumbs';
import BasePage from 'shared/components/base-page';
import Card from 'shared/components/Card';
import ClayLink from '@clayui/link';
import EmbeddedAlertList, {
	IEmbeddedAlertListProps
} from 'shared/components/EmbeddedAlertList';
import Nav from 'shared/components/Nav';
import NoResultsDisplay, {
	INoResultsDisplayProps
} from 'shared/components/NoResultsDisplay';
import React from 'react';
import SearchableEntityTable from 'shared/components/SearchableEntityTable';
import StatesRenderer from 'shared/components/states-renderer/StatesRenderer';
import URLConstants from 'shared/util/url-constants';
import {FetchSegmentsParams} from 'segment/pages/List';
import {FilterOptionType} from 'shared/types';
import {get} from 'lodash';
import {NAME} from 'shared/util/pagination';
import {Routes, toRoute} from 'shared/util/router';
import {Sizes} from 'shared/util/constants';
import {useChannelContext} from 'shared/context/channel';
import {useDataSource} from 'shared/hooks/useDataSource';
import {useParams} from 'react-router-dom';
import {User} from 'shared/util/records';

interface IBaseListPageProps {
	alerts?: IEmbeddedAlertListProps[];
	className?: string;
	columns: {
		accessor: string;
		label: any;
	}[];
	currentUser: User;
	dataSourceFn: (params: FetchSegmentsParams) => any;
	delta?: number;
	emptyStateTitle?: string;
	entityLabel?: string;
	filterBy?: object;
	filterByOptions?: FilterOptionType[];
	forwardedRef?: React.Ref<any>;
	hideNav?: boolean;
	noResultsConfig?: INoResultsDisplayProps;
	orderByOptions?: {label: string; value: string}[];
	orderIOMap?: object;
	page?: number;
	pageActions?: any[];
	pageActionsLabel?: string;
	query?: string;
	ref?: React.RefObject<SearchableEntityTable>;
	renderRowActions?: any;
	renderSelectedAction?: (checkedItemsISet) => any;
	rowIdentifier?: string;
	showCheckbox?: boolean;
}

const BaseListPage: React.FC<IBaseListPageProps> = ({
	alerts = [],
	className,
	columns,
	currentUser,
	dataSourceFn,
	delta,
	emptyStateTitle = Liferay.Language.get('no-data-sources-connected'),
	entityLabel,
	filterBy,
	filterByOptions,
	forwardedRef,
	hideNav = false,
	noResultsConfig,
	orderByOptions = [
		{
			label: Liferay.Language.get('name'),
			value: NAME
		}
	],
	orderIOMap,
	page,
	pageActions,
	pageActionsLabel,
	query,
	renderSelectedAction,
	rowIdentifier = 'id',
	showCheckbox = false,
	...otherProps
}) => {
	const {selectedChannel} = useChannelContext();
	const {channelId, groupId} = useParams();
	const authorized = currentUser.isAdmin();

	const dataSourceStates = useDataSource();
	const {empty, error, loading} = dataSourceStates;

	const renderNav = checkedItemsISet => {
		if (!hideNav && renderSelectedAction && !checkedItemsISet.isEmpty()) {
			return (
				<Nav>
					<Nav.Item key='PRIMARY_ACTION'>
						{renderSelectedAction(checkedItemsISet)}
					</Nav.Item>
				</Nav>
			);
		}
	};

	const ConnectDataSourceButton = () => (
		<ClayLink
			button
			className='button-root'
			displayType='primary'
			href={toRoute(Routes.SETTINGS_ADD_DATA_SOURCE, {
				channelId,
				groupId
			})}
		>
			{Liferay.Language.get('connect-data-source')}
		</ClayLink>
	);

	const renderNoResults = (query, activeFilters) => {
		if (query || activeFilters) {
			return (
				<NoResultsDisplay
					description={Liferay.Language.get(
						'please-try-a-different-search-term'
					)}
					icon={{
						border: false,
						size: Sizes.XXXLarge,
						symbol: 'ac-no-results-found'
					}}
					title={Liferay.Language.get('there-are-no-results-found')}
				/>
			);
		} else {
			return (
				<NoResultsDisplay
					description={get(noResultsConfig, 'description')}
					icon={get(noResultsConfig, 'icon')}
					primary
					title={get(noResultsConfig, 'title')}
				/>
			);
		}
	};

	return (
		<BasePage className={className} documentTitle={entityLabel}>
			<BasePage.Header
				breadcrumbs={[
					breadcrumbs.getHome({
						channelId,
						groupId,
						label: selectedChannel && selectedChannel.name
					})
				]}
				groupId={groupId}
			>
				<BasePage.Row>
					<BasePage.Header.TitleSection title={entityLabel} />

					<BasePage.Header.Section>
						<BasePage.Header.PageActions
							actions={pageActions}
							disabled={empty || error || loading}
							label={pageActionsLabel}
						/>
					</BasePage.Header.Section>
				</BasePage.Row>
			</BasePage.Header>

			<BasePage.Body>
				<EmbeddedAlertList alerts={alerts} />

				<Card pageDisplay>
					<Card.Body noPadding>
						<StatesRenderer {...dataSourceStates}>
							<StatesRenderer.Empty
								description={
									authorized ? (
										<>
											{Liferay.Language.get(
												'connect-a-data-source-to-get-started'
											)}

											<a
												className='d-block mb-3'
												href={
													URLConstants.DataSourceConnection
												}
												key='DOCUMENTATION'
												target='_blank'
											>
												{Liferay.Language.get(
													'access-our-documentation-to-learn-more'
												)}
											</a>

											<ConnectDataSourceButton />
										</>
									) : (
										Liferay.Language.get(
											'please-contact-your-site-administrator-to-add-people-data-sources'
										)
									)
								}
								displayCard
								title={emptyStateTitle}
							/>

							<StatesRenderer.Success>
								<SearchableEntityTable
									{...otherProps}
									columns={columns}
									dataSourceFn={dataSourceFn}
									dataSourceParams={{channelId, groupId}}
									delta={delta}
									entityLabel={entityLabel}
									filterBy={filterBy}
									filterByOptions={filterByOptions}
									navRenderer={renderNav}
									noResultsRenderer={renderNoResults}
									orderByOptions={orderByOptions}
									orderIOMap={orderIOMap}
									page={page}
									query={query}
									ref={forwardedRef}
									rowIdentifier={rowIdentifier}
									showCheckbox={showCheckbox}
								/>
							</StatesRenderer.Success>
						</StatesRenderer>
					</Card.Body>
				</Card>
			</BasePage.Body>
		</BasePage>
	);
};

export default React.forwardRef<HTMLDivElement, IBaseListPageProps>(
	(props, ref) => (
		<BaseListPage {...props} forwardedRef={ref}>
			{props.children}
		</BaseListPage>
	)
);
