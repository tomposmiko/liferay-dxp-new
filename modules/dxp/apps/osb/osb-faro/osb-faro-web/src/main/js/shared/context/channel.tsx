import React, {useContext, useReducer} from 'react';
import {Channel} from 'shared/components/channels-menu';

export enum ActionType {
	setChannels = 'setChannels',
	setSelectedChannel = 'setSelectedChannel'
}

type Action = {
	payload: any;
	type: ActionType;
};

type State = {
	channels: Array<Channel>;
	selectedChannel: Channel;
};

type ChannelProviderProps = {
	children: React.ReactNode;
	selectedChannel?: Channel;
};

export const ChannelContext = React.createContext<{
	channelDispatch?: React.Dispatch<Action>;
	channels: Array<Channel>;
	selectedChannel: Channel;
}>({
	channels: [],
	selectedChannel: null
});

export const channelReducer = (state: State, {payload, type}: Action) => {
	switch (type) {
		case ActionType.setChannels: {
			return {
				...state,
				channels: payload
			};
		}
		case ActionType.setSelectedChannel: {
			return {
				...state,
				selectedChannel: payload
			};
		}
		default:
			return state;
	}
};

export const ChannelProvider = ({
	children,
	selectedChannel: channelProp = null
}: ChannelProviderProps) => {
	const [{channels, selectedChannel}, channelDispatch] = useReducer(
		channelReducer,
		{
			channels: [],
			selectedChannel: channelProp || null
		}
	);

	return (
		<ChannelContext.Provider
			value={{channelDispatch, channels, selectedChannel}}
		>
			{children}
		</ChannelContext.Provider>
	);
};

export default ChannelProvider;

export const useChannelContext = () => useContext(ChannelContext);
