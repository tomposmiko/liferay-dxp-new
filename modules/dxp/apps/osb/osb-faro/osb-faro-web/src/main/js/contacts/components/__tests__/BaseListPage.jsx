import * as data from 'test/data';
import * as useDataSource from 'shared/hooks/useDataSource';
import BaseListPage from '../BaseListPage';
import mockStore from 'test/mock-store';
import Promise from 'metal-promise';
import React from 'react';
import {ChannelContext} from 'shared/context/channel';
import {
	cleanup,
	render,
	waitForElement,
	waitForElementToBeRemoved
} from '@testing-library/react';
import {createOrderIOMap} from 'shared/util/pagination';
import {MemoryRouter, Route} from 'react-router-dom';
import {mockChannelContext} from 'test/mock-channel-context';
import {mockEmptyState, mockSuccessState} from 'test/__mocks__/mock-objects';
import {Provider} from 'react-redux';
import {Routes} from 'shared/util/router';
import {times} from 'lodash';
import {User} from 'shared/util/records';

const TOTAL = 5;

const ACCOUNTS = times(TOTAL, i => data.mockAccount(i));

const USER = new User(data.mockUser());

const defaultProps = (empty = false) => ({
	channelId: '123123',
	columns: [
		{
			accessor: 'name',
			label: 'Name'
		},
		{
			accessor: 'id',
			label: 'Id'
		}
	],
	currentUser: USER,
	dataSourceFn: jest.fn(() =>
		Promise.resolve(
			empty ? {items: [], total: 0} : {items: ACCOUNTS, total: TOTAL}
		)
	),
	entityLabel: 'Accounts',
	groupId: '23',
	noResultsConfig: {
		description: 'There is no account data from existing data sources.',
		title: 'No account data available.'
	},
	orderIOMap: createOrderIOMap('name')
});

const WrappedComponent = ({alerts, empty, query}) => (
	<Provider store={mockStore()}>
		<MemoryRouter
			initialEntries={['/workspace/23/123123/contacts/segments']}
		>
			<Route path={Routes.CONTACTS_LIST_SEGMENT}>
				<ChannelContext.Provider value={mockChannelContext()}>
					<BaseListPage
						alerts={alerts}
						{...defaultProps(empty)}
						query={query}
					/>
				</ChannelContext.Provider>
			</Route>
		</MemoryRouter>
	</Provider>
);

jest.unmock('react-dom');
jest.useRealTimers();

const mockUseDataSource = useDataSource;

describe('BaseListPage', () => {
	afterEach(cleanup);
	mockUseDataSource.useDataSource = jest.fn(() => mockSuccessState);

	it('should render', async () => {
		const {container} = render(<WrappedComponent />);

		await waitForElementToBeRemoved(() =>
			container.querySelector('.spinner-root')
		);

		expect(container).toMatchSnapshot();
	});

	it('should load accounts', async () => {
		const {container, getByText} = render(<WrappedComponent />);

		await waitForElementToBeRemoved(() =>
			container.querySelector('.spinner-root')
		);

		expect(container.querySelectorAll('.table-head-title')).toHaveLength(2);
		expect(container.querySelector('tbody').children).toHaveLength(5);
		expect(getByText('account0')).toBeInTheDocument();
		expect(getByText('Showing 1 to 2 of 5 entries.')).toBeInTheDocument();
	});

	it('should render "No Account" empty state with no query', async () => {
		const {container, getByText} = render(<WrappedComponent empty />);

		await waitForElementToBeRemoved(() =>
			container.querySelector('.spinner-root')
		);

		expect(
			getByText('There is no account data from existing data sources.')
		).toBeInTheDocument();
		expect(getByText('No account data available.')).toBeInTheDocument();
	});

	it('should render different message in the empty state with query', async () => {
		const {container, getByText} = render(
			<WrappedComponent empty query='test' />
		);

		await waitForElementToBeRemoved(() =>
			container.querySelector('.spinner-root')
		);

		expect(container.querySelector('.tbar-nav').children).toHaveLength(2);

		expect(
			container.querySelectorAll('.tbar-section.text-truncate')[0]
		).toHaveTextContent('0 Results for "test"');
		expect(
			container.querySelectorAll('.tbar-section.text-truncate')[1]
		).toHaveTextContent('Clear');

		expect(getByText('There are no results found.')).toBeInTheDocument();
	});

	it('should render with alerts', async () => {
		const {getByRole, getByText} = render(
			<WrappedComponent alerts={[{message: 'foo alert'}]} />
		);

		await waitForElement(() => getByRole('alert'));

		expect(getByText('foo alert')).toBeInTheDocument();
	});
});

describe('BaseListPage with no Data Source', () => {
	it('should render EmptyState', () => {
		mockUseDataSource.useDataSource = jest.fn(() => mockEmptyState);

		const {getByText} = render(<WrappedComponent />);

		expect(getByText('No Data Sources Connected')).toBeInTheDocument();
		expect(
			getByText('Connect a data source to get started.')
		).toBeInTheDocument();
		expect(
			getByText('Access our documentation to learn more.')
		).toBeInTheDocument();
	});
});
