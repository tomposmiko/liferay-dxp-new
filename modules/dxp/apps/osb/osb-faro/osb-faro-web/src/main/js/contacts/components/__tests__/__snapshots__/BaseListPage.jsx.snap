// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseListPage should render 1`] = `
<div>
  <div
    class="index-root"
  >
    <div
      class="maintenance-alert-root"
    />
    <header
      class="header-root"
    >
      <div
        class="header-container"
      >
        <div
          class="row-root align-items-center d-flex justify-content-between"
        >
          <ol
            class="breadcrumb breadcrumbs-root"
          >
            <li
              class="breadcrumb-item"
            >
              <a
                class="breadcrumb-link breadcrumb-text-truncate"
                href="/workspace/23/123123/sites"
              >
                Channel 1
              </a>
            </li>
          </ol>
        </div>
        <div
          class="row-root align-items-center d-flex justify-content-between"
        >
          <div
            class="header-section title-section"
          >
            <span
              class="align-items-center d-flex"
            >
              <h1
                class="title text-truncate"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  Accounts
                </span>
              </h1>
            </span>
          </div>
          <div
            class="header-section"
          />
        </div>
      </div>
    </header>
    <div
      class="body-root page-container"
    >
      <span
        class="children-wrapper"
      >
        <div
          class="embedded-alert-list-root"
        />
        <div
          class="card card-root page-display"
        >
          <div
            class="card-body no-padding"
          >
            <div
              class="base-results-root d-flex flex-column flex-grow-1 searchable-table-root searchable-table-borderless"
            >
              <div
                class="toolbar-root"
              >
                <nav
                  class="navbar navbar-root management-bar management-bar-light navbar-expand"
                >
                  <div
                    class="container-fluid"
                  >
                    <ul
                      class="nav-root front-nav navbar-nav"
                      role="tablist"
                    >
                      <li
                        aria-haspopup="true"
                        class="dropdown dropdown-root filter-and-order-root nav-item"
                      >
                        <button
                          class="button-root dropdown-toggle nav-btn show-caret btn btn-unstyled"
                          data-testid="filter-and-order-button"
                          type="button"
                        >
                          <span
                            class="text-truncate"
                          >
                            Order
                          </span>
                          
                          <span
                            class="caret-root"
                          >
                            <svg
                              class="lexicon-icon lexicon-icon-caret-bottom icon-root"
                              role="presentation"
                            >
                              <use
                                xlink:href="#caret-bottom"
                              />
                            </svg>
                          </span>
                        </button>
                      </li>
                      <li
                        class="nav-item"
                        role="presentation"
                      >
                        <a
                          class="btn-root nav-link nav-link-monospaced btn btn-outline-borderless btn-outline-unstyled"
                          href="/?field=name&page=1&sortOrder=DESC"
                        >
                          <svg
                            class="lexicon-icon lexicon-icon-order-arrow-ascending icon-root"
                            role="presentation"
                          >
                            <use
                              xlink:href="#order-arrow-ascending"
                            />
                          </svg>
                        </a>
                      </li>
                    </ul>
                    <div
                      class="navbar-form navbar-form-autofit"
                    >
                      <div
                        class="input-group search"
                      >
                        <div
                          class="input-group-item"
                        >
                          <input
                            class="input-root form-control input-group-inset input-group-inset-after"
                            placeholder="Search"
                            type="text"
                            value=""
                          />
                          <div
                            class="input-group-inset-item input-group-inset-item-after"
                          >
                            <button
                              class="button-root btn btn-unstyled"
                              type="button"
                            >
                              <svg
                                class="lexicon-icon lexicon-icon-search icon-root"
                                role="presentation"
                              >
                                <use
                                  xlink:href="#search"
                                />
                              </svg>
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </nav>
              </div>
              <div
                class="table-responsive table-root flex-grow-1"
              >
                <table
                  class="table table-autofit table-hover table-heading-nowrap table-nowrap"
                >
                  <thead>
                    <tr>
                      <th
                        class="table-head-title"
                      >
                        <a
                          class="button-root inline-item text-truncate-inline btn btn-unstyled"
                          href="/?field=name&page=1&sortOrder=DESC"
                        >
                          <span
                            class="text-truncate"
                          >
                            Name
                          </span>
                          <span
                            class="inline-item inline-item-after"
                          >
                            <svg
                              class="lexicon-icon lexicon-icon-order-arrow-up icon-root"
                              role="presentation"
                            >
                              <use
                                xlink:href="#order-arrow-up"
                              />
                            </svg>
                          </span>
                        </a>
                      </th>
                      <th
                        class="table-head-title"
                      >
                        <a
                          class="button-root inline-item text-truncate-inline btn btn-unstyled"
                          href="/?field=id&page=1&sortOrder=ASC"
                        >
                          <span
                            class="text-truncate"
                          >
                            Id
                          </span>
                        </a>
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr
                      class=""
                    >
                      <td>
                        account0
                      </td>
                      <td>
                        0
                      </td>
                    </tr>
                    <tr
                      class=""
                    >
                      <td>
                        account1
                      </td>
                      <td>
                        1
                      </td>
                    </tr>
                    <tr
                      class=""
                    >
                      <td>
                        account2
                      </td>
                      <td>
                        2
                      </td>
                    </tr>
                    <tr
                      class=""
                    >
                      <td>
                        account3
                      </td>
                      <td>
                        3
                      </td>
                    </tr>
                    <tr
                      class=""
                    >
                      <td>
                        account4
                      </td>
                      <td>
                        4
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <nav
                class="navbar navbar-root pagination-bar-root"
              >
                <div
                  class="container-fluid"
                >
                  <div
                    aria-haspopup="true"
                    class="dropdown dropdown-root pagination-items-per-page"
                  >
                    <button
                      class="button-root dropdown-toggle show-caret btn btn-sm btn-unstyled"
                      type="button"
                    >
                      <span
                        class="text-truncate"
                      >
                        2 Items
                      </span>
                      
                      <span
                        class="caret-root"
                      >
                        <svg
                          class="lexicon-icon lexicon-icon-caret-bottom icon-root"
                          role="presentation"
                        >
                          <use
                            xlink:href="#caret-bottom"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="pagination-results"
                  >
                    Showing 1 to 2 of 5 entries.
                  </div>
                  <ul
                    class="pagination pagination-root"
                  >
                    <li
                      class="page-item disabled"
                    >
                      <a
                        class="button-root page-link btn btn-unstyled"
                        disabled=""
                        href="/?page=0"
                      >
                        <svg
                          class="lexicon-icon lexicon-icon-angle-left icon-root"
                          role="presentation"
                        >
                          <use
                            xlink:href="#angle-left"
                          />
                        </svg>
                      </a>
                    </li>
                    <li
                      class="page-item active"
                    >
                      <a
                        class="button-root page-link btn btn-unstyled"
                        href="/?page=1"
                      >
                        1
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        class="button-root page-link btn btn-unstyled"
                        href="/?page=2"
                      >
                        2
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        class="button-root page-link btn btn-unstyled"
                        href="/?page=3"
                      >
                        3
                      </a>
                    </li>
                    <li
                      class="page-item"
                    >
                      <a
                        class="button-root page-link btn btn-unstyled"
                        href="/?page=2"
                      >
                        <svg
                          class="lexicon-icon lexicon-icon-angle-right icon-root"
                          role="presentation"
                        >
                          <use
                            xlink:href="#angle-right"
                          />
                        </svg>
                      </a>
                    </li>
                  </ul>
                </div>
              </nav>
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
</div>
`;
